polonius-engine/src/output/hybrid.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2019 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! A hybrid version combining the optimized Datafrog model with the
+//! location-insensitive version.
+
+use crate::output::datafrog_opt;
+use crate::output::location_insensitive;
+use crate::output::Output;
+use facts::{AllFacts, Atom};
+
+pub(super) fn compute<Region: Atom, Loan: Atom, Point: Atom>(
+    dump_enabled: bool,
+    all_facts: AllFacts<Region, Loan, Point>,
+) -> Output<Region, Loan, Point> {
+    let lins_output = location_insensitive::compute(dump_enabled, all_facts.clone());
```

Comment:
I wonder if we should modify `location_insensitive::compute` so that it takes `&all_facts` instead, so that we don't have to clone here. From looking at the code it seems like it doesn't clone a lot of the things inside, but rather copies out itself, so it'd be overall more efficient.

Reply:
I’ll have a look at what happens if I change it and report back!

Reply:
Ok; done! In some shape or form anyway.